lib/rubocop/cop/rspec/repeated_example_group_body.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for repeated describe and context block body.
+      #
+      # @example
+      #
+      #    # bad
+      #    describe 'cool feature x' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    describe 'cool feature y' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    # good
+      #    describe 'cool feature' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    describe 'another cool feature' do
+      #      it { another_predicate }
+      #    end
+      #
+      #    # bad
+      #    context 'when case x' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    context 'when case y' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    # good
+      #    context 'when case x' do
+      #      it { cool_predicate }
+      #    end
+      #
+      #    context 'when case y' do
+      #      it { another_predicate }
+      #    end
+      #
+      class RepeatedExampleGroupBody < Cop
+        MSG = 'Repeated %<group>s block body on line(s) %<loc>s'
+
+        def_node_matcher :several_blocks?, <<-PATTERN
+          (begin (block (send _ {#{ExampleGroups::ALL.node_pattern}} ...) ...)+ )
+        PATTERN
+
+        def_node_matcher :extract_metadata, '(block (send _ _ _ $...) ...)'
+        def_node_matcher :extract_body, '(block send args $...)'
+
+        def on_begin(node)
+          return unless several_blocks?(node)
+
+          repeated_group_bodies(node).each do |repeated_group|
+            group, repeats = repeated_group.values_at(:group, :repeats)
+            add_offense(group, message: message(group, repeats))
+          end
+        end
+
+        private
+
+        def repeated_group_bodies(node)
+          node
+            .children
```

Comment:
This is quite scary. What if a method is defined, or some other statement. Won't it fail?
Not sure if this needs an example (there are some in smoke test), but is still worth checking.
If fails, please feel free to extend `spec/smoke_tests/weird_rspec_spec.rb`.

Reply:
@pirj Sorry, I did not understand your point of view, since I am not an expert in RSpec. Are you afraid of the `children` method or the implementation of `repeat_group_bodies`?

Reply:
What I mean is that example groups can contain arbitrary statements, e.g:
```ruby
RSpec.describe 'A' do
  stub_all_http_calls()
  allowed_statuses = %i[open submitted approved].freeze
  before { create(:admin) }

  describe '#load' do
    ...
  end
  describe '#load' do
    ...
  end
end
```

Will offences be detected?
I see two possible culprits, one of them is `several_blocks?` that wouldn't trigger on such a spec, and another is that `.children` would return those other statements as well.

Reply:
@pirj yeah, greate note, will check it